contracts/interchain-token-service/src/contract.rs

```diff
@@ -526,8 +510,57 @@ impl InterchainTokenService {
                     );
                 }
             }
-            Message::DeployInterchainToken(_) => {
-                todo!()
+            Message::DeployInterchainToken(DeployInterchainToken {
+                token_id,
+                name,
+                symbol,
+                decimals,
+                minter,
+            }) => {
+                ensure!(
+                    Self::token_id_config(env, token_id.clone()).is_err(),
+                    ContractError::TokenAlreadyDeployed
+                );
+
+                let token_metadata = TokenMetadata {
+                    name,
+                    symbol,
+                    decimal: decimals as u32,
+                };
+
+                ensure!(
+                    validate_token_metadata(token_metadata.clone()).is_ok(),
+                    ContractError::InvalidTokenMetaData
+                );
+
+                let minter_address = if let Some(minter) = minter {
```

Comment:
```suggestion
                let minter = if let Some(minter) = minter {
```
Avoid using type names as suffixes. You can redefine the variable instead